frontend/src/components/P2EButton.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div class="btn-wrap">
+    <!--  router button  -->
+    <router-link :to="{ name: route }" class="button" :class="size" v-if="route">
+      <span>{{ mainText }}</span>
+      <span class="sub">{{ subText }}</span>
+    </router-link>
+    <!-- normal button  -->
+    <a class="button" :class="size" v-if="targetRef">
+      <span>{{ mainText }}</span>
+      <span class="sub">{{ subText }}</span>
+    </a>
+  </div>
+</template>
+
+<script lang="ts">
+
+export default {
+  props: ['mainText', 'subText', 'disabled', 'route', 'size', 'targetRef'],
+};
+
+</script>
+
+<style scoped>
```

Comment:
A lot of styles for such component, do we need all of them? Also, where is this component used?